File: tests/fixtures/go/sample.go
package main

import "fmt"

// GoStruct is a sample struct
type GoStruct struct {
	Name string
}

// GetName returns the name
func (s *GoStruct) GetName() string {
	return s.Name
}

func main() {
	s := GoStruct{Name: "test"}
	fmt.Println(s.GetName())
}

File: tests/fixtures/typescript/sample.ts
// Sample TypeScript fixture for tree-sitter parsing tests

interface MyInterface {
  name: string;
  value: number;
}

class MyClass implements MyInterface {
  name: string;
  value: number;

  constructor(name: string, value: number) {
    this.name = name;
    this.value = value;
  }

  getName(): string {
    return this.name;
  }
}

function topLevelFunction(): void {
  const instance = new MyClass("test", 42);
  console.log(instance.getName());
}
